crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -763,9 +792,23 @@ where
     ) {
         let tx_digest = *transaction.digest();
         if let Some(qd_error) = err {
-            debug!(?tx_digest, "Failed to {action}: {}", qd_error);
-            // non-retryable failure, this task reaches terminal state for now, notify waiter.
-            quorum_driver.notify(&transaction, &Err(qd_error), old_retry_times + 1);
+            match qd_error {
```

Comment:
Turn the entire if/else into a match would be cleaner with the error handling here

Reply:
Make sense! Updated.